atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/KvsSweepProgressQueueTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.queue;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.impl.InMemoryKeyValueService;
+
+public class KvsSweepProgressQueueTest {
+    private KeyValueService kvs;
+    private KvsSweepQueueProgress progress;
+
+    @Before
+    public void setup() {
+        kvs = new InMemoryKeyValueService(true);
+        progress = new KvsSweepQueueProgress(kvs);
+    }
+
+    @Test
+    public void canReadInitialNumberOfShards() {
+        assertThat(progress.numberOfShards()).isEqualTo(1);
+    }
+
+    @Test
+    public void canUpdateNumberOfShards() {
+        progress.numberOfShards();
+        progress.updateNumberOfShards(128);
+        assertThat(progress.numberOfShards()).isEqualTo(128);
+    }
+
+    @Test
+    public void attemptingToDecreaseNumberOfShardsIsNoop() {
+        progress.numberOfShards();
+        progress.updateNumberOfShards(64);
+        progress.updateNumberOfShards(32);
```

Comment:
No exception in this case?

Reply:
Subject to change, depending on how we implement the control over this parameter. I can see throwing if we set it as an install config, but probably want to WARN and continue using the old value still.